example/app/src/main/java/com/example/mpsdkquickstart/MainActivity.kt

```diff
@@ -69,15 +71,28 @@ class MainActivity : ComponentActivity() {
     enableEdgeToEdge()
     setContent {
       MPSDKQuickstartTheme {
-        MainScreen(MobilePaymentsSdk.authorizationManager().authorizationState.isAuthorized)
-      }
-    }
+        var isAuthorized by remember { mutableStateOf(MobilePaymentsSdk.authorizationManager().authorizationState.isAuthorized) }
 
-    // Show mock reader if application is authorized and in Sandbox environment
-    if (MobilePaymentsSdk.authorizationManager().authorizationState.isAuthorized &&
-      MobilePaymentsSdk.isSandboxEnvironment()
-    ) {
-      MockReaderUI.show()
+        DisposableEffect(Unit) {
+          // Add callback and update isAuthorized from the callback
+          val callback = MobilePaymentsSdk.authorizationManager().setAuthorizationStateChangedCallback { result ->
+            isAuthorized = result.isAuthorized
+          }
+
+          // Make sure to clear the callback when the composable is disposed
+          onDispose {
+            callback.clear()
+          }
+        }
+
+        LaunchedEffect(isAuthorized) {
```

Comment:
Ah LaunchedEffect! Didn't know about this, that's what I was missing. That's more elegant than other solutions I was looking at. Thanks for taking care of this!

Reply:
ohh yeah, all the side effects in compose are a bit tricky. For this case the block within the effect will only run is isAuthorized changes. So it's a nice way to change state or do whatever you need to do once instead of in every composition.